modules/otel-troubleshoot-instrumentation.adoc

```diff
@@ -0,0 +1,91 @@
+// Module included in the following assemblies:
+//
+// * observability/otel/otel-troubleshooting.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="troubleshooting-instrumentation_{context}"]
+= Troubleshooting instrumentation
+
+You can troubleshoot instrumentation issues as follows.
+
+.Procedure
+
+== Ensure the Instrumentation was injected properly in your workload
+
+. Check the instrumentation object is created:
++
+[source,console]
+----
+oc get instrumentation -n <your-project>
+----
+
+. Verify that the resources were deployed in the correct order
++
+Deploy the Instrumentation resource before the application to ensure auto-instrumentation works correctly.
+
+When the pod starts, the operator checks for the annotation to know if it needs to inject an instrumentation. It will check for an Instrumentation and inject auto-instrumentation into the pod based on the settings specified in the annotation. Generally, this process adds an init-container to the application’s pod, which is responsible for injecting the auto-instrumentation into the app container and some environment variables.
+
+If the Instrumentation resource is not available when the application is deployed, the operator will not inject the auto-instrumentation. When you want to update the instrumentation settings, you need to delete the instrumentation object and redeploy the application.
+
+To verify that the `opentelemetry-auto-instrumentation`` init-container has successfully started, execute the following command:
+
+[source,console]
+----
+oc get events -n <your-project>
+----
+
+You should see something like this:
+
+[source,console]
+----
+... Created container opentelemetry-auto-instrumentation
+... Started container opentelemetry-auto-instrumentation
+----
+
+If those messages are missing, then it means that was an issue while injecting the auto-instrumentation.
```

Comment:
Considering this as a decision tree, what should the user know or do if these messages are OK?

Reply:
If the messages are ok:
```
[source,console]
----
... Created container opentelemetry-auto-instrumentation
... Started container opentelemetry-auto-instrumentation
----
```

Else, no messages.

Reply:
Done.